Sources/ArgumentParser/Usage/DumpHelpInfoGenerator.swift

```diff
@@ -0,0 +1,186 @@
+//===----------------------------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift Argument Parser open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+@_implementationOnly import Foundation
+
+internal struct DumpHelpInfoGenerator {
+  struct CommandInfo: Codable {
+    internal init(command: DumpHelpInfoGenerator.CommandInfo.ArgumentInfo, subcommands: [DumpHelpInfoGenerator.CommandInfo]?, arguments: [DumpHelpInfoGenerator.CommandInfo.ArgumentInfo]?, options: [DumpHelpInfoGenerator.CommandInfo.ArgumentInfo]?) {
+      self.command = command
+      self.subcommands = subcommands
+      self.arguments = arguments
+      self.options = options
+    }
+    
+    struct ArgumentInfo: Codable, Hashable {
+      internal init(name: [String]? = nil, abstract: String, discussion: String, isRequired: Bool? = nil, defaultValue: String? = nil, valueName: String? = nil, isDefault: Bool? = nil) {
+        self.name = name
+        self.abstract = abstract
+        self.discussion = discussion
+        self.isRequired = isRequired
+        self.defaultValue = defaultValue
+        self.valueName = valueName
+        self.isDefault = isDefault
+      }
+      
+      // Only for options and commands
+      var name: [String]?
+      
+      // Shared properties
+      var abstract: String
+      var discussion: String
+      
+      // Used only for arguments and options
+      var isRequired: Bool?
+      var defaultValue: String?
+      var valueName: String?
+      
+      // Used only for subcommands
+      var isDefault: Bool?
+    }
+    
+    var command: ArgumentInfo
```

Comment:
Let's not re-use this for the command's metadata — should be okay to just put the command-specific details right here instead.

Reply:
Fixed at 9c8e5a7